### ISSUE TITLE
JS/JSX Files not excluded when --type-check parameter present

### ISSUE THREAD

**Issue by harsimranb** [palantir/tslint#3069]
### Bug Report

- __TSLint version__: 5.5.0
- __TypeScript version__: 2.4.2
- __Running TSLint via__: CLI

#### Description
I have a JavaScript file with flow code that I want TSLint to ignore.  I have tried the exclude parameter of the tslint command, as well as exclude files property in the tsconfig file.  TSLint still runs lint on the JavaScript files and returns a bunch of errors.

#### JavaScript (flow) code being linted

```js
/* tslint:disable */
const isFunction = function(functionToCheck: any): boolean {
  let getType = {};
  return functionToCheck && getType.toString.call(functionToCheck) === '[object Function]';
};
module.exports = isFunction;
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-for-in-array": false,
    "no-inferred-empty-object-type": false,
    "restrict-plus-operands": false
  }
}

```

#### CLI Command
`node node_modules/tslint/bin/tslint './{components}/**/*.{ts,tsx}' --config tslint.json --exclude \"src/**/*.{js,jsx}\" --project tsconfig.json --type-check`

#### Actual behavior
TSLint ignores exclude parameters and runs lint on JS and JSX files when `--type-check` parameter present.  It returns the following errors:
```
Error at src\components\EditorContainer.js:297:16: ':' expected.
Error at src\components\utils\isFunction.js:3:46: 'types' can only be used in a .ts file.
Error at src\components\utils\isFunction.js:3:52: 'types' can only be used in a .ts file.
```

#### Expected behavior
JS and JSX files should be ignored by TSLint when exclude parameter is present.

**Comment by ajafff** [comment]
Closing as a duplicate of #1603

You can solve this problem by omitting `--type-check`. It's no longer needed to enable typed rules.
